### PR TITLE
Add rst documentation for deprecated enums

### PR DESCRIPTION
**Pull request by andreas-bok-sociomantic**
- Add missing rst documentation for drizzle_socket_option_t
- Replace drizzle_socket_owner with drizzle_socket_owner_t

### CODE REVIEW

File: docs/api/constants.rst [hunk 1]
```diff
@@ -741,14 +743,42 @@ Connection
...
 
    .. py:data:: DRIZZLE_SOCKET_OWNER_NATIVE
 
    .. py:data:: DRIZZLE_SOCKET_OWNER_CLIENT
 
+.. c:type:: (deprecated) drizzle_socket_owner
+
+   typedef of :c:type:`drizzle_socket_owner_t`
+
+.. c:type:: drizzle_socket_option_t
```
Comment (ben-palmer-sociomantic) [inline docs/api/constants.rst:756]: Should this documentation also be required in `5.x.x`?

File: docs/api/constants.rst [hunk 2]
```diff
@@ -716,11 +716,12 @@ Connection
 
       Verify SSL cert
 
-   .. py:data:: DRIZZLE_CAPABILITIES_REMEBER_OPTIONS
+   .. py:data:: DRIZZLE_CAPABILITIES_REMEMBER_OPTIONS
```
Comment (ben-palmer-sociomantic) [inline docs/api/constants.rst]: I would separate the fixes and the additions in to two commits.

**Review by ben-palmer-sociomantic — COMMENTED** [review]
Couple of minor comments.

File: docs/api/constants.rst (hunk 1 above)
Comment (andreas-bok-sociomantic) [reply docs/api/constants.rst:756]: Yeah, just checked. It is in this commit 94cff771a4524c3fc4438487202c3a613eecf355 on v5.x.x

**Comment by andreas-bok-sociomantic** [comment]
Updated based on comments from @ben-palmer-sociomantic 
Also rebased on v5.x.x since the deprecations were added on this branch and not merged into v6.x.x yet

**Comment by andreas-bok-sociomantic** [comment]
Moved the commit into changes for v5.x.x to a new PR (#208)

**Comment by ben-palmer-sociomantic** [comment]
LGTM

**Pull request merged**